src/zkp/pisch.rs

```diff
@@ -189,11 +187,17 @@ impl PiSchProof {
 mod tests {
     use super::*;
     use crate::{utils::testing::init_testing, zkp::BadContext};
+    fn transcript() -> Transcript {
+        Transcript::new(b"PiSchProof Test")
+    }
+
+    type TestFn = fn(PiSchProof, PiSchInput) -> Result<()>;
 
-    fn random_schnorr_proof<R: RngCore + CryptoRng>(
+    fn with_random_schnorr_proof<R: RngCore + CryptoRng>(
         rng: &mut R,
         additive: bool,
```

Comment:
thought: I don't really like this notation, it's not a good name and it's not clear when you call tests what it's doing, but that is not a problem for this PR. I should probably add an issue to #52.